SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/BaseObject.cpp

```diff
@@ -254,9 +254,19 @@ void BaseObject::init()
 {
     for(auto data: this->m_vecData)
     {
-        if (data->isRequired() && !data->isSet())
+        if (data->isRequired())
         {
-            msg_error() << "Required data \"" << data->getName() << "\" has not been set. (Current value is " << data->getValueString() << ")" ;
+            if(!data->isSet())
+            {
+                if(data->hasDefaultValue())
+                {
+                    msg_info() << "Required data \"" << data->getName() << "\" has not been set but has a default value. (Current value is " << data->getValueString() << ")" ;
+                }
+                else
+                {
+                    msg_error() << "Required data \"" << data->getName() << "\" has not been set and has not default value. ";
+                }
+            }
```

Comment:
If isRequired and hasDefaultValue are set to true by default, this is a bit extreme in terms of verbosity... 
I agree with this approach, but now I think isRequired() should be false by default. There are much more optional data than required ones.

Reply:
@damienmarchal what do you think?

Reply:
Ouch we are already March 1.